main.go

```diff
@@ -125,14 +125,17 @@ func (t *Test) TheApplicationIsRunningInStaging(statusCode int) {
 		}
 		if u == "" {
 			return fmt.Errorf("No URL found for environment %s", key)
+			utils.LogInfo("still looking for app env info url")
 		}
 		return nil
 	}
-	err := RetryExponentialBackoff(time.Minute*10, f)
-	Expect(err).ShouldNot(HaveOccurred(), "get applications with a URL")
+	err := RetryExponentialBackoff(time.Minute * 10, f)
+	if err != nil {
```

Comment:
Why do you need to check for the error being non-nil before the `Expect`? Doesn't this defeat the point of the `Expect`?

Reply:
i was seeing a NPE here occasionally, the best i could determine it was happening due to the `err.Error()` call in the message.  I can retest without those changes and see if i can reproduce.

Reply:
Yeah, that would make sense.
The code flow would be (I think):
1) Call `Expect(err)`, passing in `nil`. Store result as a temporary variable
2) evaluate `HaveOccurred()`
3) evaluate `err.Error()`    <---- NPE here
4) call `ShouldNot(...)` on the result of 1, passing in the result of 2, `"get applications with..."` and the result of 3

So AFAIUI, even if `ShouldNot` is a no-op, the `err.Error()` is getting evaluated and a NRE getting thrown.